lib/dialects/postgres/query-generator.js

```diff
@@ -897,6 +897,13 @@ var QueryGenerator = {
       return 'ARRAY[' + value.map(function (v) {
         return SqlString.escape(JSON.stringify(v), false, this.options.timezone, this.dialect, field);
       }, this).join(',') + ']::' + jsonType.key + '[]';
+    } else if (value && field && field.type instanceof DataTypes.GEOMETRY) {
+      var s = "ST_GeomFromGeoJSON('" + JSON.stringify(value) + "'";
+      if (field.type.srid) {
+        s += + ", " + field.type.srid;
```

Comment:
Thanks for the heads up. I'll be submitting a fix for this issue tomorrow
to address the following bug -
https://github.com/sequelize/sequelize/issues/4054. If you need things to
work for now, you can leave off the SRID. Thanks again.

Mike Ringrose

On Tue, Jul 7, 2015 at 9:22 PM, Kris Rasmussen <notifications@github.com>
wrote:

> In lib/dialects/postgres/query-generator.js
> <https://github.com/sequelize/sequelize/pull/4013#discussion_r34108922>:
>
> > @@ -897,6 +897,13 @@ var QueryGenerator = {
> >        return 'ARRAY[' + value.map(function (v) {
> >          return SqlString.escape(JSON.stringify(v), false, this.options.timezone, this.dialect, field);
> >        }, this).join(',') + ']::' + jsonType.key + '[]';
> > +    } else if (value && field && field.type instanceof DataTypes.GEOMETRY) {
> > +      var s = "ST_GeomFromGeoJSON('" + JSON.stringify(value) + "'";
> > +      if (field.type.srid) {
> > +        s += + ", " + field.type.srid;
>
> hey @mikeringrose <https://github.com/mikeringrose>, I think you are
> either missing another term in this statement, or need to remove the
> leading "+". I tried creating geometry with DataTypes.GEOMETRY("Point",
> 4326) and ended up having bad queries generated due to this line.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sequelize/sequelize/pull/4013/files#r34108922>.
>